clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2960,16 +2960,21 @@ class SYCLKernelNameTypeVisitor
     if (T.isNull())
       return;
     const CXXRecordDecl *RD = T->getAsCXXRecordDecl();
+
     if (!RD) {
       if (T->isNullPtrType()) {
-        S.Diag(KernelInvocationFuncLoc, diag::err_sycl_kernel_incorrectly_named)
-            << KernelNameType;
-        S.Diag(KernelInvocationFuncLoc, diag::note_invalid_type_in_sycl_kernel)
-            << /* kernel name cannot be a type in the std namespace */ 2 << T;
+        S.Diag(KernelInvocationFuncLoc,
+               diag::err_invalid_std_type_in_sycl_kernel)
+            << KernelNameType << T->getAsTagDecl();
+
         IsInvalid = true;
       }
       return;
     }
+    // if (!RD && T->isTypedefNameType()) {
+    //   const auto *TDefType = T->getAs<TypedefType>();
+    //    VisitTypedefType(TDefType);
+    // }
```

Comment:
Unneeded code.

Reply:
This is still a draft and not ready for formal review.
@AaronBallman and I are discussing about adding additional Visitor methods, which motivated adding this new Visitor code.